sdk/httputil/http.go

```diff
@@ -0,0 +1,64 @@
+package httputil
+
+import (
+	"crypto/tls"
+	"net/http"
+	"time"
+)
+
+const (
+	defaultTimeout = 120 * time.Second
+	// defaults to match DefaultTransport - defined to satisfy lint
+	maxIdleConns          = 100
+	idleConnTimeout       = 90 * time.Second
+	tLSHandshakeTimeout   = 10 * time.Second
+	expectContinueTimeout = 1 * time.Second
+)
+
+var preventRedirectCheck = func(_ *http.Request, _ []*http.Request) error {
+	return http.ErrUseLastResponse // Prevent following redirects
```

Comment:
Completely preventing redirects may not be realistic.  However I am hopeful, because determining what IDOR's are allowed can be challenging.

I am proposing this initially, but it need more testing. If we are uncomfortable with this, maybe we can add some logging or metrics that we could use to help validate the safety here?

Reply:
Is this something where an end-user might want to support following redirects? Or maybe an allowlist of redirect endpoints? The latter seems more complicated than adding `sdk.WithUnsafeFollowRedirects()`

Reply:
> Is this something where an end-user might want to support

Not unless we add an allow list / allowed IDOR configuration.  Without that configuration this feature is insecure and could be used to make internal requests with the help of a redirect.

Since configuration adds complexity, I think we should only add it if we find it's necessary. There should not be a large need for this, so it would be good to understand that need first.